network/network_windows.go

```diff
@@ -63,10 +63,20 @@ func (nm *networkManager) newNetworkImplHnsV1(nwInfo *NetworkInfo, extIf *extern
 	)
 
 	networkAdapterName := extIf.Name
+
+	// Pass adapter name here if it is not empty, this is cause if we don't tell HNS which adapter to use
+	// it will just pick one randomly, this is a problem for customers that have multiple adapters
+	if nwInfo.AdapterName != "" {
+		networkAdapterName = nwInfo.AdapterName
+	}
+
 	// FixMe: Find a better way to check if a nic that is selected is not part of a vSwitch
 	if strings.HasPrefix(networkAdapterName, vEthernetAdapterPrefix) {
 		networkAdapterName = ""
```

Comment:
can you add a comment here indicating that HNS returns error if the supplied adapter name contains vEthernet prefix?
also it would good to add a log indicating setting the adapter name to empty.
This will help debug in cases, when user supplies vEthernet adapter name, we will set to correctly on line 70 but will reset back to empty on 75. Having a log will help debug such issues and comment in the code will help us know why we cannot supply vEthernet adapter name to HNS.

Reply:
Added both a comment and log